source/MaterialXContrib/MaterialXMaya/ReloadMaterialXLibrariesCmd.cpp

```diff
@@ -0,0 +1,50 @@
+#include "ReloadMaterialXLibrariesCmd.h"
+#include "Plugin.h"
+
+#include <maya/MGlobal.h>
+#include <maya/MSyntax.h>
+
+#include <MaterialXCore/Document.h>
+
+namespace MaterialXMaya
+{
+
+MString ReloadMaterialXLibrariesCmd::NAME("reloadMaterialXLibraries");
+
+ReloadMaterialXLibrariesCmd::ReloadMaterialXLibrariesCmd()
+{
+}
+
+ReloadMaterialXLibrariesCmd::~ReloadMaterialXLibrariesCmd()
+{
+}
+
+MStatus ReloadMaterialXLibrariesCmd::doIt(const MArgList& /*args*/)
+{
+    try
+    {
+        Plugin::instance().loadLibraries();
```

Comment:
As far as I can see this will update the cached library document member but not otherwise do anything else. Seems only CreateMaterialXNodeCmd::doit() and MaterialXNode::createAndRegisterFragment() actually use the cached value. Thus any adverse affect of changing the library document will not occur until either of those entry points get's called: I think this is create new or modifying existing (on reload).